### ISSUE TITLE
Comparison to pytorch version

### ISSUE THREAD

**Issue by pr4deepr** [mukund-ks/DeepLabV3-Segmentation#4]
Hi
  Thanks for setting this library up.
I was trying this on some of my data for a binary segmentation task, and tested both the tensorflow and [pytorch](https://github.com/mukund-ks/DeepLabV3Plus-PyTorch) version of deeplabv3plus. It seemed that the tensorflow version was performing better. 
Are there any changes in architecture that may make a difference? I could see a change in dilations for ASPP and a tanh activation layer in the pytorch version.

Thanks

Pradeep

**Comment by mukund-ks** [comment]
Hi,
The amount of dilation layers in the ASPP module could account for the difference in metrics between Tensorflow and PyTorch-implemented models. The PyTorch model has just four Atrous Convolution layers (rates as, 6, 12, 18, and 24), but the Tensorflow model has five (rates as, 6, 12, 18, 24, and 36). Another cause for this scenario could be backend optimizations between the two libraries, however I'm not certain. Weight initialization is also important; the Tensorflow model makes use of He initialization.

Regarding the activation differences between the two implementations, utilizing Leaky ReLU and Normalized Tanh was simply an experiment. For my specific task, I didn't notice any substantial improvements in segmentation quality or metrics.

The PyTorch implementation requires an upgrade, which I will make soon.

Cheers!
Mukund

**Comment by pr4deepr** [comment]
Thanks @mukund-ks , appreciate that.

I'll experiment a bit more and post some information here.

Looking forward to the upgrade!

Cheers
Pradeep

**Comment by mukund-ks** [comment]
Hi @pr4deepr ,
The necessary changes to the PyTorch implementation have been added. Please refer to [DeepLabV3Plus-PyTorch@69d85a6](https://github.com/mukund-ks/DeepLabV3Plus-PyTorch/commit/69d85a686e80854489c3b63cfbe44d125c644143) for details.

This comment closes the issue.
Thanks!
